### ISSUE TITLE
Navbar ordering and grouping

### ISSUE THREAD

**Issue by kMutagene** [fsprojects/FSharp.Formatting#628]
It seems like the documents in the sidebar {{fsdocs-list-of-documents}} is just ordering the files according to the OS file order. 

consider i have the following files in the `content` folder

The only thing these files contain are h1 headers that together form the sentence 

`Some More Content So You Can See The Hopefully Nice Scrolling Behaviour Of This Menu Unless You Have The Highest Screen Ever`

```
	0_Markdown-Cheatsheet.md
    1_fsharp-code-example.fsx
    a1.fsx -> "Some"
    a2.fsx -> "More"
    a3.fsx -> "Content"
	etc. ...
```

The order on windows is correct, while it is messed up on ubuntu as you can see [here](https://fslab.org/docs-template/content/0_Markdown-Cheatsheet.html) - i guess due to differences in file ordering. 

Additionally, it may be of high interest to give the possibility to not only order the documentation content, but also create groups in the sidebar, e.g. 

```
Documentation
Group A
	DocumentA1
	DocumentA2
Group B 
	DocumentB1
	DocumentB2
	Group BA
		Document BA1
```

Ideas that i would have on how to approach this:

- Extend the `LiterateDocModel` to allow at least file index and (nested) groups

- Build the `LiterateDocModel` from frontmatter in the documents, like for example jekyll (which uses YAML frontmatter). Grouping and ordering could then look something like this:

	  (***front-matter***) // or any other kind of indicator like --- in .md files
	  document-title: document title
	  document-index: 1 // or start counting at 0 idk
	  document-group: some/path/that/will/be/translated/by/fsdocs

- if there is no frontmatter, or keys are missing, fall back to default behavior
- alternatively to defining the `document-group` in the frontmatter, just follow the file structure in the `docs` folder, e.g. 
```
Some/Path/file.fsx` ->
Some
	Path
		file

```

Is this of interest?

**Comment by kMutagene** [comment]
Small update on this, i am currently using something similar to the frontmatter suggested above in a project that combines fornax and FSharp.Formatting. To make fsdocs ignore the frontmatter, it is wrapped in normal comments. 

https://github.com/fslaborg/fslabsite/blob/main/src/content/tutorials_src/4_getting-started.fsx

```fsharp
(***hide***)

(*
#frontmatter
---
title: Getting started
category: datascience
authors: David Zimmer
index: 0
---
*)
```

I realized during this that free-form usage of frontmatter might not be a good fit for FSharp.Formatting, as it is usually used for highly variable key value pairs, not for a small set. Static site generators that allow for custom scripts when generating the webpages like fornax can leverage this to its full extend, but fsdocs tries to take that complexity away. 

So long story short i think this should be done in the style of available literate commands. The only two things needed would be `(***document-group:<GROUP/PATH/AS/SHOWN/ABOVE>***)` and `(***document-group-index:<INDEX>***)` with the following behavior:

- When both are not present, act as the group is `/` (same as now) and append it to the end of the sorted section of `/`
- When only the group is present, append it to the end of the sorted section of that group
- When only the index is present, act as the group is `/` and use the index for sorting
- When both are present, use the index for sorting in the respective group

**Comment by dsyme** [comment]
> So long story short i think this should be done in the style of available literate commands. The only two things needed would be (***document-group:<GROUP/PATH/AS/SHOWN/ABOVE>***) and (***document-group-index:<INDEX>***) with the following behavior:

I guess so, or else maybe something like this?  However I don't mind too much either way

(***frontmatter
---
title: Getting started
category: datascience
authors: David Zimmer
index: 0
---
***)

**Comment by dsyme** [comment]
@kMutagene It seems to me we can just do

(***
---
title: Getting started
category: datascience
authors: David Zimmer
index: 0
---
***)

and then apply usual rules of markdown processing determine that this is frontmatter? This would then apply to both markdown and .fsx inputs?

**Comment by dsyme** [comment]
@kMutagene Could you cast an eye over https://github.com/fsprojects/FSharp.Formatting/pull/666 and try it out?

thanks
don

**Comment by dsyme** [comment]
Note we're not taking `authors` into account, nor using the frontmatter as template substitutions.  We could do that I suppose

**Comment by kMutagene** [comment]
@dsyme looking into it right now! Regarding authors, I do not think that that is too important, the frontmatter itself was more of a random example from the fslab website.

**Comment by kMutagene** [comment]
@dsyme It works like a charm, very cool! Regarding nesting of these categories, i am not sure if that would be even needed anymore, as most likely anything above 2 levels would look strange anyways and the ordering via both indexes seems enough for me as is.

**Comment by dsyme** [comment]
Yup working well for me!  I'm really happy about this, thanks for making the suggestion - I'm not sure I would have iterated to a pleasant solution anywhere near fast enough if I'd had to solve this myself (I likely would have over-engineered it)

![image](https://user-images.githubusercontent.com/7204669/114931100-b017d180-9e2d-11eb-8203-b31b79232488.png)

**Comment by kMutagene** [comment]
Happy that i could help here. This library becomes even more useful with every new release, it is really cool!

**Comment by dsyme** [comment]
Yeah I think we're getting somewhere.   I'll go do some other work now though :-)

**Comment by dsyme** [comment]
I tweeted about recent improvements here: https://twitter.com/dsymetweets/status/1382324615968661506
